### PR TITLE
[WIP] Add a draft write-up for the release

### PR DESCRIPTION
**Pull request by Stebalien**
This should cover most of the breaking changes but doesn't cover some of the new
features.

Still need to flesh out the zombie streams documentation (e.g., FullClose etc.)

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+get stuck on open behind a queue of connect events all being handled in series.
+In theory, these events should have been handled quickly but in practice, it's
+very hard to avoid locks *entirely* (bitswap's event handlers were especially
+problematic).
+
+Worse, this serial delivery guarantee didn't actually provide us with an
+*in-order* delivery guarantee as it was still possible for a disconnect to
+happen before we even *started* to fire the connect event. The situation was
+slightly better than before because the events couldn't overlap but still far
+from optimal.
```
Comment (whyrusleeping) [inline NEWS.md]: maybe try to end this section on a happier note. "This has all been resolved... and should be super good now"

Comment (Stebalien) [reply NEWS.md]: Yeah, the entire thing could use a bit more "pep" (I was tired).

File: NEWS.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+What does this mean for end users? Well:
+
+1. Reference counting connections to a peer using connect/disconnect events
+   should "just work" and should never go negative.
+2. Under heavy connect/disconnect loads, connecting to new peers should be
+   faster (usually).
+
+In the past, (dis)connect and stream open/close notifications have been a bit of
+a pain point. For a long time, they were fired of in parallel and one could, for
```
Comment (bigs) [inline NEWS.md]: fired off

File: NEWS.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
+# go-libp2p release notes
+
+## 6.0.0
+
+We're pleased to announce go-libp2p 6.0.0. This release includes a massive
+refactor of go-libp2p that paves the way for new transports such as QUICK.
+Unfortunately, as it is a sweeping change, there are some breaking changes,
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply NEWS.md]: Isn't that bit redundant?

[the remark this answers is not in the document]
Comment (bigs) [reply NEWS.md]: ahhh sorry... meant to have it read like this:

> Unfortunately, as the refactor is broad sweeping, there are some breaking changes

to eliminate the change change thing. super minor, was just reading for style i guess haha

File: NEWS.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+recommend that you try again. A simple example can be found in the
+[echo](https://github.com/libp2p/go-libp2p/tree/feat/refactor/examples/echo)
+example.
+
+Given this work and in an attempt to consolidate all of our configuration logic
+in one place, we've removed all default transports from go-libp2p-swarm.
+
+TL;DR: Please use the libp2p constructor.
+
+### Zombie Streams
```
Comment (Stebalien) [inline NEWS.md]: @bigs @mgoelzer 

This is a bit of an annoying libp2p quirk you should know about. We may eventually add some way to automatically garbage collect zombie streams but the lack of weak references in go makes this a bit difficult.

File: NEWS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mgoelzer) [reply NEWS.md]: Minor nit, but it should be `new transports such as QUIC` (no K).

Really excited for this PR!

File: NEWS.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+
+#### Zombie Streams
+
+From this release on, when you're done with a stream, you must either call
+`Reset` on it (in case of an error) or close it and read an EOF (or some other
+error). Otherwise, libp2p can't determine if the stream is *actually* closed and
+will hang onto it indefinitely.
+
+To make properly closing streams a bit easier, we've added two methods to
+[go-libp2p-net][]://godoc.org/github.com/libp2p/go-libp2p-net): `AwaitEOF`
```
Comment (upperwal) [inline NEWS.md]: typo

**Comment by Stebalien** [comment]
@mgoelzer I believe this is ready to go. I've fleshed it out significantly so I'd appreciate any additional feedback.

File: NEWS.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+[go-libp2p-swarm]: https://github.com/libp2p/go-libp2p/go-libp2p-swarm
+[go-libp2p-swarm/testing]: https://github.com/libp2p/go-libp2p/go-libp2p-swarm/testing
+[go-libp2p-transport]: https://github.com/libp2p/go-libp2p-transport
+[go-libp2p-transport/test]: https://github.com/libp2p/go-libp2p-transport/test
+[go-libp2p-transport-upgrader]: https://github.com/libp2p/go-libp2p-transport-upgrader
+[go-peerstream]: https://github.com/libp2p/go-peerstream
+[go-tcp-transport]: https://github.com/libp2p/go-tcp-transport
+[go-ws-transport]: https://github.com/libp2p/go-ws-transport
+
+[example:echo]: https://github.com/libp2p/go-libp2p/tree/feat/refactor/examples/echo
```
Comment (mgoelzer) [inline NEWS.md]: Should be:  https://github.com/libp2p/go-libp2p/tree/master/examples/echo

**Comment by mgoelzer** [comment]
LGTM.  I think the level of detail / historical context is great, and also thanks for fixing all the examples (I had missed that until today).

**Comment by whyrusleeping** [comment]
@mgoelzer do you want to push this out as a blog post? Libp2p doesnt have its own blog right now, but i'd be happy to have it out on the ipfs blog

**Pull request merged**